src/ArborX_LinearBVH.hpp

```diff
@@ -137,8 +137,11 @@ class BoundingVolumeHierarchy
 };
 
 template <typename DeviceType>
-class BoundingVolumeHierarchy<
-    DeviceType, std::enable_if_t<Kokkos::is_device<DeviceType>::value>>
+// clang-format off
+class [[deprecated("Use BoundingVolumeHierarchy<MemorySpace> instead.")]]
```

Comment:
This is more about the class template, than a function. Both constructor and query would need to be mentioned if you want to expand. I'm open to changing this comment to something better.

Reply:
My train of thought was more one will get the warning and change the tree type, then try to compile again and get an error because the old query overload does not exist.

Reply:
I am not asking to change anything.  Just thinking out loud.

Reply:
```suggestion
class [[deprecated("Use BoundingVolumeHierarchy<MemorySpace> instead and use an execution space as first argument in 'query'.")]]
```